bluesky/run_engine.py

```diff
@@ -924,6 +944,55 @@ def _cache_config(self, obj):
         self._config_ts_cache[obj] = config_ts
 
     @asyncio.coroutine
+    def _monitor(self, msg):
+        if not self._run_is_open:
+            raise IllegalMessageSequence("A 'monitor' message was sent but no "
+                                         "run is open.")
+        if obj in self._monitor_cbs:
+            raise IllegalMessageSequence("A 'monitor' message was sent for {}"
+                                         "which is already monitored".format(
+                                             obj))
+
+        descriptor_uid = new_uid()
+        data_keys = obj.describe()
+        config = obj.read_configuration()
+        object_keys = {obj.name: list(data_keys)}
+        doc = dict(run_start=self._run_start_uid, time=ttime.time(),
+                   data_keys=data_keys, uid=descriptor_uid,
+                   configuration=config, name=None,
+                   object_keys=object_keys)
+        yield from self.emit(DocumentNames.descriptor, doc)
+        self.log.debug("Emitted Event Descriptor")
+
+        def emit_event(*args, **kwargs):
+            # Ignore the inputs. Use this call as a signal to call read on the
+            # object, a crude way to be sure we get all the info we need.
+
+            data, timestamps = _rearrange_into_parallel_dicts(obj.read())
```

Comment:
Does everything in `monitor_attrs` have to be a signal? It can't be a subdevice?